workspaces/src/network/variants.rs

```diff
@@ -75,6 +77,11 @@ where
     pub async fn dev_generate(&self) -> (AccountId, SecretKey) {
         let id = crate::rpc::tool::random_account_id();
         let sk = SecretKey::from_seed(KeyType::ED25519, DEV_ACCOUNT_SEED);
+        if self.info().name.eq("testnet") {
+            let id: AccountId = AccountId::from_str(format!("{}.testnet", id.as_str()).as_str())
+                .expect("could not convert string account id into AccountId");
+            return (id, sk);
+        }
```

Comment:
This is not a maintainable way of addressing it. I cannot approve this PR with a workaround. Consider putting this code somewhere in the testnet dev_create_account/url_create_account implementation.